src/calculators/BinaryVotingPowerEarningPowerCalculator.sol

```diff
@@ -0,0 +1,210 @@
+  );
+
+  /// @notice Emitted when a new eligibility module is set.
+  /// @param oldEligibilityModule The old eligibility module address.
+  /// @param newEligibilityModule The new eligibility module address.
+  event EligibilityModuleSet(
+    address indexed oldEligibilityModule, address indexed newEligibilityModule
+  );
+
+  /*///////////////////////////////////////////////////////////////
+                          Errors
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Thrown when an invalid address is provided.
+  error BinaryVotingPowerEarningPowerCalculator__InvalidAddress();
+
+  /// @notice Thrown when an invalid voting power update interval is provided.
+  error BinaryVotingPowerEarningPowerCalculator__InvalidVotingPowerUpdateInterval();
+
+  /*///////////////////////////////////////////////////////////////
+                        Immutable Storage
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice The voting power token address.
+  address public immutable VOTING_POWER_TOKEN;
+
+  /// @notice The block number at which the voting power update starts.
+  uint48 public immutable SNAPSHOT_START_BLOCK;
+
+  /*///////////////////////////////////////////////////////////////
+                          Storage
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice The time interval after which the voting power is updated.
+  uint48 public votingPowerUpdateInterval;
+
+  /// @notice The oracle eligibility module that determines if a delegate qualifies for earning
+  /// power.
+  IOracleEligibilityModule public oracleEligibilityModule;
+
+  /*///////////////////////////////////////////////////////////////
+                            Constructor
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Initializes the BinaryVotingPowerEarningPowerCalculator.
+  /// @param _owner The owner of the contract.
+  /// @param _oracleEligibilityModule The oracle eligibility module address.
+  /// @param _votingPowerToken The voting power token address.
+  /// @param _votingPowerUpdateInterval The voting power update interval.
+  constructor(
+    address _owner,
+    address _oracleEligibilityModule,
+    address _votingPowerToken,
+    uint48 _votingPowerUpdateInterval
+  ) Ownable(_owner) {
+    if (_votingPowerToken == address(0)) {
+      revert BinaryVotingPowerEarningPowerCalculator__InvalidAddress();
+    }
+
+    VOTING_POWER_TOKEN = _votingPowerToken;
+    SNAPSHOT_START_BLOCK = uint48(block.number);
+
+    _setOracleEligibilityModule(_oracleEligibilityModule);
+    _setVotingPowerUpdateInterval(_votingPowerUpdateInterval);
+  }
+
+  /*///////////////////////////////////////////////////////////////
+              Proxy Eligibility Module Functions
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Gets the earning power of a voter.
```

Comment:
Expand on this, explain that it is a combination of the staker voting power and the stakers eligibility.

Reply:
Same for getNewEarningpower